external/corefx-bugfix/src/Common/src/Interop/Windows/kernel32/Interop.WIN32_FIND_DATA.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class Kernel32
+    {
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        internal unsafe struct WIN32_FIND_DATA
+        {
+            internal uint dwFileAttributes;
+            internal FILE_TIME ftCreationTime;
+            internal FILE_TIME ftLastAccessTime;
+            internal FILE_TIME ftLastWriteTime;
+            internal uint nFileSizeHigh;
+            internal uint nFileSizeLow;
+            internal uint dwReserved0;
+            internal uint dwReserved1;
+            private fixed char _cFileName[MAX_PATH];
+            private fixed char _cAlternateFileName[14];
+
+            internal ReadOnlySpan<char> cFileName
+            {
+                get { fixed (char* c = _cFileName) return new ReadOnlySpan<char>(c, MAX_PATH); }
+            }
+
+#if UNITY_AOT
+            internal void SetFileName(string fileName)
+            {
+                fixed (char* c = fileName)
+                {
+                    for (var i = 0; i < fileName.Length; i++)
+                        _cFileName[i] = c[i];
```

Comment:
There's a potential buffer overflow here, you need to check if fileName.Length is less than MAX_PATH.

Reply:
Good point! I've made a fix and force-pushed it. Please have another look.